src/main/java/com/configcat/ConfigCatClient.java

```diff
@@ -340,11 +340,47 @@ public void clearDefaultUser() {
         this.defaultUser = null;
     }
 
+    @Override
+    public boolean isClosed() {
+        return isClosed.get();
+    }
+
     @Override
     public void close() throws IOException {
+        if (!this.isClosed.compareAndSet(false, true)) {
+            return;
+        }
+        synchronized (INSTANCES){
+            if(INSTANCES.containsKey(sdkKey)){
```

Comment:
`get()` will eventually work as a contains check as it returns `null` when the key doesn't exist in the map. I think it would be enough to use check against the result of `get()`.